frontend/src/store/FamLoginUserState.ts

```diff
@@ -0,0 +1,168 @@
+import { AdminMgmtApiService } from '@/services/ApiServiceFactory';
+import { CURRENT_SELECTED_APPLICATION_KEY } from '@/store/ApplicationState';
+import { FAM_APPLICATION_NAME } from '@/store/Constants';
+import { setRouteToastError } from '@/store/ToastState';
+import type { CognitoUserSession } from 'amazon-cognito-identity-js';
+import { AdminRoleAuthGroup, type FamApplicationDto, type FamAuthGrantDto } from 'fam-admin-mgmt-api/model';
+import { readonly, ref } from "vue";
+
+const FAM_LOGIN_USER = 'famLoginUser';
+
+// For some of the values of the FAMLoginUser properties, see AuthService.parseToken().
+export interface FamLoginUser {
+    username?: string;
+    displayName?: string;
+    email?: string;
+    idpProvider?: string; // from ID Token's ['identities']['providerName'] attribute.
+    roles?: string[]; // roles from Access Token's ['cognito:groups']. This may soon be redundant after delegated admin design.
+    authToken?: CognitoUserSession; // original JWT token from AWS Cognito (ID && Access Tokens).
+    accesses?: FamAuthGrantDto[]; // admin privileges retrieved from backend.
+};
+
+const state = ref({
+    famLoginUser: localStorage.getItem(FAM_LOGIN_USER)
+        ? (JSON.parse(localStorage.getItem(FAM_LOGIN_USER) as string) as
+              | FamLoginUser
+              | undefined
+              | null)
+        : undefined,
+});
+
+// --- getters
+
+const getAuthToken = () => {
+    return state.value.famLoginUser?.authToken;
+};
+
+const getUserAccess = () => {
+    return state.value.famLoginUser?.accesses;
+};
+
+// admin levels: 'FAM_ADMIN', 'APP_ADMIN', 'DELEGATED_ADMIN'
+const getUserAdminRoleGroups = () => {
+    const accesses = getUserAccess();
+    if (accesses && accesses.length > 0) {
+        return accesses.map(
+            (access) => access.auth_key
+        )
+    }
+};
+
+/**
+ * @param adminRole three admin levels: 'FAM_ADMIN', 'APP_ADMIN', 'DELEGATED_ADMIN'
+ * @returns array of applications the login user has been granted for
+ *          administering on based on adimn level.
+ */
+const getApplicationsAdministeredByAdminRole = (
```

Comment:
Hi Ian, just personal thoughts... I wonder if we can just call it getAppsByAdminRole.... the applications administered is too long I feel....

Reply:
Sure, actually I like the short name initially but was afraid it is not so clear to others.
I will use the shorter name. 
I created all methods that are related and close to the state file to provide functionalities and would be easier to know where they are first before even touch components but either places are fine to me. I will move to the component.

Reply:
oh sorry I just saw you use it in getApplicationsUserAdministers method as well, it confused me a bit.... I might prefer to separate it, so this call will just be used by fam admin

Reply:
Yes, those methods are useful as they are very closed related to the state. That's why it is much clear to me to be in the container of the same file and just provide them for whoever needs them (even just for one component, much like util may just be used for one function).
If we separate them out into components then component will be packed with complicated code that they are not suppose to be, and basic common logic will also need to be repeated in both places.

Reply:
I'm fine to keep it here, but maybe restrict it to call used by fam admin, and in the getApplicationsUserAdministers method, just go through the whole access list, and check `access.grants.forEach`, then you don't need to reuse this method, and feel it's more clear